R/endpoints.R

```diff
@@ -233,31 +234,44 @@ endpoint_validate_baseline_combined <- function(req, res, pjnz, shape, populatio
 #' @return Bytes of zip file.
 #' @keywords internal
 endpoint_download_spectrum <- function(queue) {
-  function(res, result, id) {
-    download(res, result, id, "spectrum")
-  }
+  download(queue, "spectrum")
 }
 
-#' Download indicator zip file.
+#' Download summary zip file.
 #'
 #' Returns a function which returns bytes of zip.
 #'
 #' @param queue The queue used to retrieve download from task id.
 #'
 #' @return Bytes of zip file.
 #' @keywords internal
-endpoint_download_indicators <- function(queue) {
-  function(res, result, id) {
-    download(res, result, id, "indicators")
-  }
+endpoint_download_summary <- function(queue) {
+  download(queue, "summary")
 }
 
-download <- function(res, result, id, file) {
-  ## TODO: Get path to download from the queue using ID and file type
-  ## stream this back mrc-624
-  ## with_success(queue$result(id)[[file]])
-  path <- system_file("output", "malawi.zip")
-  readBin(path, "raw", n = file.info(path)$size)
+download <- function(queue, type) {
+  function(req, res, id) {
+    response <- with_success(queue$result(id))
+    if (is_error(response$value)) {
+      response$success <- FALSE
+      response$errors <- hintr_errors(
+        list("MODEL_RUN_FAILED" = scalar(toString(response$value)))
```

Comment:
That's because if the model run fails it returns an `rrq_task_error` condition. Probably better to explicitly pull the message from that though instead of relying on `toString`. I've updated that.